comb_spec_searcher/rule_db/base.py

```diff
@@ -57,10 +65,10 @@ def add(self, start: int, ends: Iterable[int], rule: AbstractRule) -> None:
         is_equiv = len(ends) == 1 and rule.is_two_way()
         if is_equiv:
             self.set_equivalent(start, ends[0])
-        if len(ends) != 1 or is_equiv or not self.are_equivalent(start, ends[0]):
-            # to avoid overwriting an equivalence rule with a non-equivalence
-            # rule, we only save if an equivalence rule, or does not have the
-            # # same start -> ends as some equivalence rule.
+            self.eqv_rule_to_strategy[(start, ends)] = rule.strategy
+            self.rule_to_strategy.pop((start, ends), None)
+            self.rule_to_strategy.pop((ends[0], (start,)), None)
+        else:
```

Comment:
Should we only add if we know that start and ends are not equivalent?

Reply:
I can do that